build/templates/session.py.mako

```diff
@@ -297,15 +235,17 @@ class Session(_SessionBase):
         ${p['python_name']} = _converters.${p['python_api_converter_name']}(${p['python_name']})
 %   endif
 % endfor
-        self._library = _library_singleton.get()
-        self._encoding = 'windows-1251'
+        self._library_interpreter = _library_interpreter.LibraryInterpreter(encoding='windows-1251')
 
         # Call specified init function
-        self._${config['session_handle_parameter_name']} = 0  # This must be set before calling ${init_function['python_name']}().
-        self._${config['session_handle_parameter_name']} = self.${init_function['python_name']}(${init_call_params})
+        # Note that _library_interpreter sets _${config['session_handle_parameter_name']} to 0 in its constructor, so that if
+        # ${init_function['python_name']} fails, the error handler can reference it.
+        # And then once ${init_function['python_name']} succeeds, we can update _library_interpreter._${config['session_handle_parameter_name']}
+        # with the actual session handle.
+        self._library_interpreter._${config['session_handle_parameter_name']} = self.${init_function['python_name']}(${init_call_params})
```

Comment:
Regarding the session handle, see my comment in the other conversation.

Regarding caching LibraryInterpreter as a singleton, I don't think that would be a good change. It would add complexity (another class) for not enough benefit. For the Library class, it's important because we don't want to load the CDLL functions more than we have to. But LibraryInterpreter doesn't do anything worth caching.